ddtrace/ext/ci.py

```diff
@@ -41,28 +43,35 @@
 
 
 def _normalize_ref(name):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_TAGS.sub("", _RE_ORIGIN.sub("", _RE_REFS.sub("", name))) if name is not None else None
 
 
 def _filter_sensitive_info(url):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_URL.sub("\\1", url) if url is not None else None
 
 
 def tags(env=None):
+    # type: (Optional[MutableMapping[str, str]]) -> MutableMapping[str, str]
+    """Extract and set tags from provider environ."""
     env = os.environ if env is None else env
-    tags = {}
+    tags = {}  # type: MutableMapping[str, Optional[str]]
```

Comment:
any reason to do `MutableMapping` instead of `Dict`?

I haven't seen `MutableMapping` typing before, not sure when/why I should be using this

(this is personal curiosity question, not suggesting a change)

Reply:
It turns out `env` is a `MutableMapping` instead of a `Dict`. (MutableMapping also apparently the abstract base class for Dict) So since this function is taking the `env` and updating with the extracted info from CI and git, I had to make sure they were compatible ☹️

Reply:
That's actually a great point, I missed that the first time. I'll change the return type signatures accordingly :D